src/component/project/ProjectAccess/ProjectAccessAddUser/ProjectAccessAddUser.tsx

```diff
@@ -0,0 +1,241 @@
+        setOptions([]);
+        if (typeof selectedUser === 'string' || selectedUser === null) {
+            return;
+        }
+
+        if (selectedUser?.id) {
+            setUser(selectedUser);
+        }
+    };
+
+    const handleRoleChange = (
+        evt: React.ChangeEvent<{
+            name?: string | undefined;
+            value: unknown;
+        }>
+    ) => {
+        const roleId = Number(evt.target.value);
+        const role = roles.find(role => role.id === roleId);
+        if (role) {
+            setRole(role);
+        }
+    };
+
+    const handleSubmit = async (evt: React.SyntheticEvent) => {
+        evt.preventDefault();
+        if (!role || !user) {
+            setToastData({
+                type: 'error',
+                title: 'Invalid selection',
+                text: `The selected user or role does not exist`,
+            });
+            return;
+        }
+
+        try {
+            await addUserToRole(id, role.id, user.id);
+            refetchProjectAccess();
+            setUser(undefined);
+            setOptions([]);
+            setToastData({
+                type: 'success',
+                title: 'Added user to project',
+                text: `User added to the project with the role of ${role.name}`,
+            });
+        } catch (e: any) {
+            let error;
+
+            if (
+                e
+                    .toString()
+                    .includes(`User already has access to project=${id}`)
+            ) {
+                error = `User already has access to project ${id}`;
+            } else {
+                error = e.toString() || 'Server problems when adding users.';
+            }
+            setToastData({
+                type: 'error',
+                title: error,
+            });
+        }
+    };
+
+    return (
+        <>
+            <Alert severity="info" style={{ marginBottom: '20px' }}>
+                The user must have an Unleash root role before added to the
+                project.
+            </Alert>
+            <Grid container spacing={3} alignItems="flex-end">
+                <Grid item>
+                    <Autocomplete
+                        id="add-user-component"
+                        style={{ width: 300 }}
+                        noOptionsText="No users found."
+                        onChange={handleSelectUser}
+                        onBlur={() => handleBlur()}
+                        value={user || ''}
+                        freeSolo
+                        getOptionSelected={() => true}
+                        filterOptions={o => o}
+                        getOptionLabel={(option: IProjectAccessUsers) => {
+                            if (option) {
+                                return `${option.name || '(Empty name)'} <${
+                                    option.email || option.username
+                                }>`;
+                            } else return '';
+                        }}
```

Comment:
Would put this fn in a `const` outside the `return`, for readability.

Reply:
Good point. I'll fix it.